### PR TITLE
Update to CSV data extractor to permit finding file in either app or plugin

### PR DESCRIPTION
**Pull request by redxeth**
Minor fix-- to let Origen file_handler help find the CSV file.

### CODE REVIEW

File: lib/origen/utility/csv_data.rb [hunk 1]
```diff
@@ -77,7 +77,8 @@ def extract_csv_data(options = {}) # :nodoc:
         field_values = []
         result = []
         line_no = 1
-        File.readlines("#{Origen.root}/#{@file}").each do |line|
+        file = Origen.file_handler.clean_path_to(@file)
+        File.readlines("#{@file}").each do |line|
           # Clean up line
```
Comment (ginty) [inline lib/origen/utility/csv_data.rb]: Doesn't look right to me, updates `file` but then goes on to reference `@file`...

Comment (redxeth) [reply lib/origen/utility/csv_data.rb]: Thanks, copy typo.  Strange that it worked anyway for my 2 tests cases with the bug.  test cases : file in plugin, file in toplevel app. Fixed.

**Comment by info-rchitect** [comment]
@redxeth With all of the CSV import gems are we sure we are using the best method here by rolling our own?  Maybe that is a better question outside this PR but I am asking because I get requests to import CSV all the time and have not had an answer for 'Origen recommends'.  thx

**Comment by redxeth** [comment]
@info-rchitect  Yeah the CSV import capability was added a while ago-- don't have any checkin logs as it was before origen (rgen) was on github.  Anyway for now just using it and needed a little fix.

**Comment by info-rchitect** [comment]
LGTM

**Pull request merged**